backend/corpus_asset_pipelines/integrated_corpus/load.py

```diff
@@ -0,0 +1,98 @@
+import logging
+import time
+from typing import List
+
+import numpy as np
+import pandas as pd
+import tiledb
+
+from backend.corpus_asset_pipelines.integrated_corpus.transform import transform_dataset_raw_counts_to_rankit
+from backend.wmg.data.schemas.corpus_schema import obs_labels, var_labels, VAR_ARRAY_NAME, OBS_ARRAY_NAME
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def load_dataset(corpus_path: str, anndata_object: pd.DataFrame, dataset_id: str):
+    """
+    Read given anndata dataset into the tiledb object (under corpus name), updating the var and feature indexes
+    to avoid collisions within the larger tiledb object
+    """
+    var_df = update_corpus_var(corpus_path, anndata_object)
+
+    # Calculate mapping between var/feature coordinates in anndata_object (local h5ad file) and corpus (global tdb object) # noqa E501
+    global_var_index = np.zeros((anndata_object.shape[1],), dtype=np.uint32)
+    var_feature_to_coord_map = {k: v for k, v in var_df[["gene_ontology_term_id", "var_idx"]].to_dict("split")["data"]}
+    for idx in range(anndata_object.shape[1]):
+        gene_ontology_term_id = anndata_object.var.index.values[idx]
+        global_coord = var_feature_to_coord_map[gene_ontology_term_id]
+        global_var_index[idx] = global_coord
+
+    first_obs_idx = update_corpus_obs(corpus_path, anndata_object, dataset_id)
+    start = time.time()
```

Comment:
For instrumenting code, `time_ns` is the preferred function since it gives a higher resolution time.

Reply:
Once https://github.com/chanzuckerberg/single-cell-data-portal/pull/2794 I'll remove the timer and use the decorator function (using `time.perf_counter`)